lib/Page.js

```diff
@@ -775,20 +775,20 @@ class Page extends EventEmitter {
    * @param {!Array<string>} values
    */
   async select(selector, ...values) {
-    await this.$eval(selector, (element, values) => {
+    return await this.$eval(selector, (element, values) => {
       if (element.nodeName.toLowerCase() !== 'select')
         throw new Error('Element is not a <select> element.');
-
       const options = Array.from(element.options);
-
+      const filteredValues = values.filter(value => options.some(option => option.value === value));
       if (element.multiple) {
         for (const option of options)
-          option.selected = values.includes(option.value);
+          option.selected = filteredValues.includes(option.value);
```

Comment:
I don't understand why you need `filteredValues`. This line can be rewritten as the following:
```js
option.selected = values.includes(option.value);
```